neuro_san/http_sidecar/handlers/base_request_handler.py

```diff
@@ -203,14 +205,32 @@ def data_received(self, chunk):
     def prepare(self):
         self.logger.info(self.get_metadata(), f"[REQUEST RECEIVED] {self.request.method} {self.request.uri}")
 
-    def set_default_headers(self):
-        # Force to close this connection without trying to reuse it.
-        self.set_header("Connection", "close")
+    def do_finish(self):
+        """
+        Wrapper for finish() call
+        with check for closed client connection.
+        """
+        try:
+            self.finish()
+        except tornado.iostream.StreamClosedError:
+            self.logger.warning(self.get_metadata(), "Finish: client closed connection unexpectedly.")
+
+    async def do_flush(self) -> bool:
+        """
+        Wrapper for flush() call
+        with check for closed client connection.
+        """
+        try:
+            await self.flush()
+            return True
+        except tornado.iostream.StreamClosedError:
+            self.logger.warning(self.get_metadata(), "Flush: client closed connection unexpectedly.")
+            return False
 
     async def options(self, *_args, **_kwargs):
         """
         Handles OPTIONS requests for CORS support
         """
         # No body needed. Tornado will return a 204 No Content by default
         self.set_status(http.HTTPStatus.NO_CONTENT)
-        await self.finish()
+        self.do_finish()
```

Comment:
Here and below - doing wrapper for handler finish() instead of flush() makes more sense,
if this is not a streaming request.